packages/components/src/utils/useDebounce.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useMemo, useRef } from "react";
+import debounce from "lodash/debounce";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type AnyFunction = (...args: any[]) => any;
+
+/**
+ * A hook to easily manage debounced functions, including their cleanup.
+ * @param func The function to debounce.
+ * @param wait The number of milliseconds to delay.
+ * @param options Optional debounce settings.
+ * @returns The debounced function.
+ */
+export function useDebounce<T extends AnyFunction>(
```

Comment:
Appreciate all the comments in here! And I agree - I think it's a good idea to keep the hook where it is for now, rather than put it into the hooks package 👍